src/Akka.Persistence.MongoDb.Tests/AtLeastOnceDeliveryActorSpecs.cs

```diff
@@ -0,0 +1,146 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Akka.Actor;
+using Akka.Actor.Dsl;
+using Akka.Configuration;
+using Akka.Event;
+using Akka.Util.Internal;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.MongoDb.Tests
+{
+
+    public class AtLeastOnceDeliveryActorSpecs : Akka.TestKit.Xunit2.TestKit, IClassFixture<DatabaseFixture>
+    {
+        public static readonly AtomicCounter Counter = new AtomicCounter(0);
+        private readonly ITestOutputHelper _output;
+
+        public AtLeastOnceDeliveryActorSpecs(ITestOutputHelper output, DatabaseFixture databaseFixture)
+            : base(CreateSpecConfig(databaseFixture, Counter.GetAndIncrement()), "AtLeastOnceDeliveryActorSpecs", output)
+        {
+            _output = output;
+            output.WriteLine(databaseFixture.ConnectionString + Counter.Current);
+        }
+
+        [Fact]
+        public void MongoDb_should_persist_AtLeastOnceDelivery_snapshots_correctly()
+        {
+            var aldProps = Props.Create(() => new AldActor("fuber", TestActor));
+            var ald1 = Sys.ActorOf(aldProps, "ald1");
+            Watch(ald1);
+            ald1.Tell("hi");
+            ald1.Tell("there");
+
+            // make sure initial payload got sent through
+            ReceiveN(2).Cast<Delivery>().All(x => x.Msg is string).Should().BeTrue();
+            Sys.Stop(ald1);
+            ExpectTerminated(ald1);
+
+            var ald2 = Sys.ActorOf(aldProps, "ald2");
+            ExpectMsg("recovered");
+            var msgs = ReceiveN(2).Cast<Delivery>();
+            foreach (var msg in msgs)
+            {
+                ald2.Tell(msg.ConfirmationId);
+            }
+            ExpectNoMsg(1000);
+        }
+
+        class Delivery
+        {
+            public Delivery(object msg, long confirmationId)
+            {
+                Msg = msg;
+                ConfirmationId = confirmationId;
+            }
+
+            public object Msg { get; }
+
+            public long ConfirmationId { get; }
+        }
+
+        class AldActor : AtLeastOnceDeliveryActor
+        {
+            private IActorRef _targetActor;
+            private readonly ILoggingAdapter _log = Context.GetLogger();
+
+            public AldActor(string persistenceId, IActorRef targetActor)
+            {
+                PersistenceId = persistenceId;
+                _targetActor = targetActor;
+            }
+
+            protected override bool ReceiveRecover(object message)
+            {
+                switch (message)
+                {
+                    case SnapshotOffer offer when offer.Snapshot is AtLeastOnceDeliverySnapshot snapshot:
+                        _targetActor.Tell("recovered");
+                        foreach (var s in snapshot.UnconfirmedDeliveries)
+                        {
+                            _log.Info("Have delivery scheduled to {0} with Id {1}", s.Destination, s.DeliveryId);
```

Comment:
Throws an NRE error - there's something structurally wrong with the `ActorPath` that is deserialized here. Haven't figured out if it's a serialization error (garbage in) or a deserialization error (garbage out) issue yet.